Server/src/main/resources/static/js/editor.js

```diff
@@ -175,6 +175,53 @@ function parseDWInstruction(line, result, currentLine) {
     }
 }
 
+function isLabel(text, result) {
+    for (i = 0; i < result.labels.length; i++) {
+        if (text === result.labels[i]) {
+            return true;
+        }
+    }
+}
+
+function isImmediateOp(text, result) {
+    return (isLabel(text, result) 
+        || !isNaN(Number(text)) 
+        && Number(text) === Math.floor(Number(text)) 
+        && text.indexOf("o") === -1
+        && text.indexOf("0e") !== 0);
+}
+
+function isMemoryOp(text) {
+    return text.startsWith("[") && text.endsWith("]");
+}
+
+function isRegisterOp(text) {
+    return new RegExp('^(a|b|c|d|x|y|bp|sp)$').test(text.toLowerCase());
+}
+
+function getOffsetOperandType(text) {
+    
+    var offset = "";
```

Comment:
It's not necessary to assign a default value to `offset` since it's assigned a value in all paths